src/lib.rs

```diff
@@ -6,6 +6,26 @@
 //! * [`Mutex`] - a mutual exclusion lock.
 //! * [`RwLock`] - a reader-writer lock, allowing any number of readers or a single writer.
 //! * [`Semaphore`] - limits the number of concurrent operations.
+//!
+//! ## Relationship with `std::sync`
+//!
+//! In general, you should prefer to use [`std::sync`] types over types from this crate.
```

Comment:
> you should

Personally, I do not believe that the benefits of using std::sync or parking_lot here necessarily exceed the risk of accidentally holding the guard across the await point.

Especially when using parking_lot, it is easy to make that mistake because of the send guard. (I recently saw a colleague make that mistake and spend hours debugging.)

Reply:
> Personally, I do not believe that the benefits of using std::sync or parking_lot here necessarily exceed the risk of accidentally holding the guard across the await point.

For standard library mutexes, there is a Clippy lint nowadays that prevents this scenario from happening. I've made the language less impactful, but I still think this isn't as large of an issue

Reply:
Maybe link to the Clippy lint? https://rust-lang.github.io/rust-clippy/stable/index.html#/await_holding_lock

Reply:
Good idea, I've added this